.github/workflows/docs.yml

```diff
@@ -0,0 +1,42 @@
+name: Auto-Generate Docs
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  generate-javadoc:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v3
+
+    - name: Set up JDK 20
+      uses: actions/setup-java@v3
+      with:
+        java-version: '20'
+
+    - name: Create docs directory if it doesn't exist
+      run: mkdir -p ./docs
+
+    - name: Generate Javadoc
+      run: mvn javadoc:javadoc       
+
+    - name: Switch to docs-deployment branch
+      run: |
+        git fetch origin
```

Comment:
I am not quite sure here... I have proposed another solution in which I just delete the docs-deployment branch and create a new clean one, with the current state (this way no merge conflicts come up and as this is only docs I think this should not be a problem). Afterwards I also delete the java files so that it only includes the docs ( and some Licence and README files). Is this fine with you or do you have a better approach in mind?